build-tools-internal/src/main/groovy/elasticsearch.run-ccs.gradle

```diff
@@ -40,9 +44,9 @@ tasks.register("run-ccs", RunTask) {
     queryingCluster.get().getNodes().each { node ->
       if (proxyMode) {
         node.setting('cluster.remote.my_remote_cluster.mode', 'proxy')
-        node.setting('cluster.remote.my_remote_cluster.proxy_address', "\"${fulfillingCluster.get().getAllTransportPortURI().get(0)}\"")
+        node.setting('cluster.remote.my_remote_cluster.proxy_address', "\"${fulfillingCluster.get().getAllRemoteAccessPortURI().get(0)}\"")
```

Comment:
I tried running ccs, but it fails with exception:

<details><summary>Unable to open any proxy connections to cluster [my_remote_cluster]</summary>

```
[2023-02-23T00:26:05,098][WARN ][o.e.t.RemoteClusterService] [querying-cluster-0] failed to connect to remote clusters java.util.concurrent.ExecutionException: org.elasticsearch.transport.NoSeedNodeLeftException: Unable to open any proxy connections to cluster [my_remote_cluster]
        at org.elasticsearch.server@8.8.0-SNAPSHOT/org.elasticsearch.action.support.PlainActionFuture$Sync.getValue(PlainActionFuture.java:310)
        at org.elasticsearch.server@8.8.0-SNAPSHOT/org.elasticsearch.action.support.PlainActionFuture$Sync.get(PlainActionFuture.java:284)
        at org.elasticsearch.server@8.8.0-SNAPSHOT/org.elasticsearch.action.support.PlainActionFuture.get(PlainActionFuture.java:73)
        at org.elasticsearch.server@8.8.0-SNAPSHOT/org.elasticsearch.transport.RemoteClusterService.initializeRemoteClusters(RemoteClusterService.java:352)
        at org.elasticsearch.server@8.8.0-SNAPSHOT/org.elasticsearch.transport.TransportService.doStart(TransportService.java:328)
        at org.elasticsearch.server@8.8.0-SNAPSHOT/org.elasticsearch.common.component.AbstractLifecycleComponent.start(AbstractLifecycleComponent.java:43)
        at org.elasticsearch.server@8.8.0-SNAPSHOT/org.elasticsearch.node.Node.start(Node.java:1353)
        at org.elasticsearch.server@8.8.0-SNAPSHOT/org.elasticsearch.bootstrap.Elasticsearch.start(Elasticsearch.java:438)
        at org.elasticsearch.server@8.8.0-SNAPSHOT/org.elasticsearch.bootstrap.Elasticsearch.initPhase3(Elasticsearch.java:231)
        at org.elasticsearch.server@8.8.0-SNAPSHOT/org.elasticsearch.bootstrap.Elasticsearch.main(Elasticsearch.java:67)
Caused by: org.elasticsearch.transport.NoSeedNodeLeftException: Unable to open any proxy connections to cluster [my_remote_cluster]
        at org.elasticsearch.server@8.8.0-SNAPSHOT/org.elasticsearch.transport.ProxyConnectionStrategy.openConnections(ProxyConnectionStrategy.java:287)
        at org.elasticsearch.server@8.8.0-SNAPSHOT/org.elasticsearch.transport.ProxyConnectionStrategy$1.onFailure(ProxyConnectionStrategy.java:255)
        at org.elasticsearch.server@8.8.0-SNAPSHOT/org.elasticsearch.transport.ProxyConnectionStrategy.lambda$openConnections$10(ProxyConnectionStrategy.java:281)
        at org.elasticsearch.server@8.8.0-SNAPSHOT/org.elasticsearch.action.ActionListener$DelegatingActionListener.onFailure(ActionListener.java:188)
        at org.elasticsearch.server@8.8.0-SNAPSHOT/org.elasticsearch.action.ActionListener$Delegating.onFailure(ActionListener.java:97)
        at org.elasticsearch.server@8.8.0-SNAPSHOT/org.elasticsearch.action.support.ContextPreservingActionListener.onFailure(ContextPreservingActionListener.java:38)
        at org.elasticsearch.server@8.8.0-SNAPSHOT/org.elasticsearch.action.ActionListener$Delegating.onFailure(ActionListener.java:97)
        at org.elasticsearch.server@8.8.0-SNAPSHOT/org.elasticsearch.action.ActionListener.completeWith(ActionListener.java:512)
        at org.elasticsearch.server@8.8.0-SNAPSHOT/org.elasticsearch.common.util.concurrent.ListenableFuture.notifyListenerDirectly(ListenableFuture.java:125)
        at org.elasticsearch.server@8.8.0-SNAPSHOT/org.elasticsearch.common.util.concurrent.ListenableFuture.done(ListenableFuture.java:114)
        at org.elasticsearch.server@8.8.0-SNAPSHOT/org.elasticsearch.action.support.PlainActionFuture.setException(PlainActionFuture.java:169)
        at org.elasticsearch.server@8.8.0-SNAPSHOT/org.elasticsearch.common.util.concurrent.ListenableFuture.onFailure(ListenableFuture.java:153)
        at org.elasticsearch.server@8.8.0-SNAPSHOT/org.elasticsearch.transport.ClusterConnectionManager.failConnectionListener(ClusterConnectionManager.java:388)
        at org.elasticsearch.server@8.8.0-SNAPSHOT/org.elasticsearch.transport.ClusterConnectionManager.lambda$connectToNodeOrRetry$5(ClusterConnectionManager.java:261)
        at org.elasticsearch.server@8.8.0-SNAPSHOT/org.elasticsearch.action.ActionListener$3.onFailure(ActionListener.java:304)
        at org.elasticsearch.server@8.8.0-SNAPSHOT/org.elasticsearch.action.ActionListener$Delegating.onFailure(ActionListener.java:97)
        at org.elasticsearch.server@8.8.0-SNAPSHOT/org.elasticsearch.action.ActionListener$RunAfterActionListener.onFailure(ActionListener.java:411)
        at org.elasticsearch.server@8.8.0-SNAPSHOT/org.elasticsearch.action.ActionListener$7.onFailure(ActionListener.java:562)
        at org.elasticsearch.server@8.8.0-SNAPSHOT/org.elasticsearch.action.ActionListener$Delegating.onFailure(ActionListener.java:97)
        at org.elasticsearch.server@8.8.0-SNAPSHOT/org.elasticsearch.action.ActionListener$Delegating.onFailure(ActionListener.java:97)
        at org.elasticsearch.server@8.8.0-SNAPSHOT/org.elasticsearch.action.ActionListenerResponseHandler.handleException(ActionListenerResponseHandler.java:48)
        at org.elasticsearch.server@8.8.0-SNAPSHOT/org.elasticsearch.transport.TransportService$ContextRestoreResponseHandler.handleException(TransportService.java:1382)
        at org.elasticsearch.server@8.8.0-SNAPSHOT/org.elasticsearch.transport.TransportService$ContextRestoreResponseHandler.handleException(TransportService.java:1382)
        at org.elasticsearch.server@8.8.0-SNAPSHOT/org.elasticsearch.transport.InboundHandler.doHandleException(InboundHandler.java:429)
        at org.elasticsearch.server@8.8.0-SNAPSHOT/org.elasticsearch.transport.InboundHandler$3.doRun(InboundHandler.java:421)
        at org.elasticsearch.server@8.8.0-SNAPSHOT/org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:958)
        at org.elasticsearch.server@8.8.0-SNAPSHOT/org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:26)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
        at java.base/java.lang.Thread.run(Thread.java:1589)
```
</details>

I'm guessing the `remote_cluster.port` is not ready yet.

Reply:
I would bet this is because we haven't configured the remote access API key for the QC. This is rather difficult with just Gradle scripts. I wonder whether we can also migrate this script to use the new inline `ElasticsearchCluster`? @mark-vieira Could you please provide us some pointers on whether this is possible (`ElasticsearchCluster` is in a different package) and if so how it can be achieved? Thanks!

Reply:
Indeed, CCS will fail without the remote access API key in QC settings 👍 

Since `run-ccs` is really just an internal util for us, I think I want to move forward with the following, and add a more automated solution (e.g., migrating to using inline `ElasticsearchCluster`) later:

* add a new flag to toggle behavior between legacy and new model (as @slobodanadamovic suggests)
* for the new model, add a `println` along the lines of "You must manually set up the remote access API key on the two clusters"

LMKWYT!

Reply:
>  I think I want to move forward with the following, and add a more automated solution (e.g., migrating to using inline ElasticsearchCluster) later

Addressing this later, sounds good to me.

> for the new model, add a `println` along the lines of "You must manually set up the remote access API key on the two clusters"

Not needed to do it now, but I was wondering if it would also make sense to document how to run and configure new model in our [TESTING](https://github.com/elastic/elasticsearch/blob/main/TESTING.asciidoc#manually-testing-cross-cluster-search) doc? Some small example which can be copy pasted could be helpful.

Reply:
Good call! I think we should do this once we're closer to the finish line and we have the right setup locked down.

Reply:
> I would bet this is because we haven't configured the remote access API key for the QC. This is rather difficult with just Gradle scripts. I wonder whether we can also migrate this script to use the new inline `ElasticsearchCluster`? @mark-vieira Could you please provide us some pointers on whether this is possible (`ElasticsearchCluster` is in a different package) and if so how it can be achieved? Thanks!

A "run" task based on the new framework doesn't currently exist but could. I suspect you're envisioning somethere were you define these clusters in Java classes somewhere as "templates" and then spin them up? I'd have to have a think about how that might work. The change here is that the new test clusters code is no longer on the build classpath, so we'd have to wire things up a little different for it to work.

Reply:
> I suspect you're envisioning somethere were you define these clusters in Java classes somewhere as "templates" and then spin them up?

Yes that's what I had in mind. It is by no means urgent, but would be nice to have it eventually. Thanks!